File: src/main/java/com/rtomyj/yugiohAPI/model/HateoasLinks.java
package com.rtomyj.yugiohAPI.model;

import java.util.List;

public interface HateoasLinks
{

    void setSelfLink();
    void setLinks();

    public static <T extends HateoasLinks> void setLinks(List<T> instances)
    {

        instances.forEach(item -> item.setLinks());

    }

}

File: src/main/kotlin/com/rtomyj/skc/util/HateoasLinks.kt
package com.rtomyj.skc.util

interface HateoasLinks {
    fun setSelfLink()
    fun setLinks()

    companion object {
        @JvmStatic
        fun setLinks(instances: List<HateoasLinks>) {
            instances
                .forEach { it.setLinks() }
        }
    }
}